src/ui/components/atoms/side-menu-button/SideMenuButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useState } from 'react';
+import type { IProps } from './types/IProps';
+import styles from './styles.module.scss';
+import Icon1 from '../../../../assets/icons/article-management.svg?raw';
+import Icon2 from '../../../../assets/icons/author-management.svg?raw';
+
+function SideMenuButton({ children, icon, color, variant, onClick }: IProps) {
+  const [isActive, setIsActive] = useState(false);
+
+  const handleClick = () => {
+    setIsActive(!isActive);
+    if (onClick) {
+      onClick();
+    }
+  };
+
+  const selectedIcon = icon ?? (children === 'Article Management' ? Icon1 : Icon2);
+
+  return (
+    <button
+      data-testid="side-menu-button-styles"
+      className={`${styles['side-menu-button']} ${styles[color ?? '']} ${
+        variant ? styles[variant] : ''
+      } ${isActive ? styles.active : ''}`}
```

Comment:
this logic will be difficult to maintain in the future. please use this approach https://github.com/Ditmar/backoffice-openscience/blob/master/src/ui/components/demo/calculator/atoms/display.tsx#L7-L10